glyphs-reader/src/font.rs

```diff
@@ -125,129 +230,340 @@ struct RawFont {
     pub manufacturerURL: Option<String>,
     pub versionMajor: Option<i64>,
     pub versionMinor: Option<i64>,
-    pub axes: Option<Vec<Axis>>,
+    pub axes: Vec<Axis>,
     pub glyphs: Vec<RawGlyph>,
     pub font_master: Vec<RawFontMaster>,
-    #[fromplist(default)]
     pub instances: Vec<RawInstance>,
-    pub feature_prefixes: Option<Vec<RawFeature>>,
-    pub features: Option<Vec<RawFeature>>,
-    pub classes: Option<Vec<RawFeature>>,
-    pub properties: Option<Vec<RawName>>,
+    pub feature_prefixes: Vec<RawFeature>,
+    pub features: Vec<RawFeature>,
+    pub classes: Vec<RawFeature>,
+    pub properties: Vec<RawName>,
+    #[fromplist(alt_name = "kerning")]
+    pub kerning_LTR: Kerning,
+    pub custom_parameters: CustomParameters,
+}
+
+#[derive(Clone, Default, Debug, PartialEq, Eq, Hash)]
+pub(crate) struct CustomParameters(BTreeMap<String, CustomParameterValue>);
+
+impl CustomParameters {
+    fn int(&self, name: &str) -> Option<i64> {
+        let Some(CustomParameterValue::Int(i)) = self.0.get(name) else {
+            return None;
+        };
+        Some(*i)
+    }
 
-    #[fromplist(rest)]
-    pub other_stuff: BTreeMap<String, Plist>,
+    fn bool(&self, name: &str) -> Option<bool> {
+        self.int(name).map(|v| v == 1)
+    }
+
+    fn string(&self, name: &str) -> Option<&str> {
+        let Some(CustomParameterValue::String(str)) = self.0.get(name) else {
+            return None;
+        };
+        Some(str)
+    }
+
+    fn axes(&self) -> Option<&Vec<Axis>> {
+        let Some(CustomParameterValue::Axes(axes)) = self.0.get("Axes") else {
+            return None;
+        };
+        Some(axes)
+    }
+
+    fn axis_mappings(&self) -> Option<&Vec<AxisMapping>> {
+        let Some(CustomParameterValue::AxesMappings(mappings)) = self.0.get("Axis Mappings") else {
+            return None;
+        };
+        Some(mappings)
```

Comment:
just an observation, but as written this is very... loose feeling. I assume that it would be Quite Bad if this key existed but had a different type? If that is true should we just unwrap? Ideally we would be validating during parse time, and then it would be an invariant that if this key exists the type is correct.

Reply:
That's fair, will fix.